gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/common/MongoFactory.java

```diff
@@ -139,28 +139,27 @@ private ServerSettings buildServerSettings() {
     private SslSettings buildSslSettings() {
         SslSettings.Builder sslBuilder = SslSettings.builder();
 
-        Boolean sslEnabled = readPropertyValue(propertyPrefix + "sslEnabled", Boolean.class);
-
-        if (sslEnabled != null) {
-            sslBuilder.enabled(sslEnabled);
-            if (sslEnabled.booleanValue()) {
-                String keystore = readPropertyValue(propertyPrefix + "keystore", String.class);
-                String keystorePassword = readPropertyValue(propertyPrefix + "keystorePassword", String.class);
-                String keyPassword = readPropertyValue(propertyPrefix + "keyPassword", String.class);
-
-                if (keystore != null) {
-                    try {
-                        SSLContext ctx = SSLContext.getInstance("TLS");
-                        KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
-                        KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
-                        ks.load(new FileInputStream(keystore), keystorePassword.toCharArray());
-                        keyManagerFactory.init(ks, keyPassword.toCharArray());
-                        ctx.init(keyManagerFactory.getKeyManagers(), null, null);
-                        sslBuilder.context(ctx);
-                    } catch (Exception e) {
-                        logger.error(e.getCause().toString());
-                        throw new IllegalStateException("Error creating the keystore for mongodb", e);
-                    }
+        Boolean sslEnabled = readPropertyValue(propertyPrefix + "sslEnabled", Boolean.class, false);
+        sslBuilder.enabled(sslEnabled);
+
+        if (sslEnabled) {
+            String keystore = readPropertyValue(propertyPrefix + "keystore", String.class);
+            String keystorePassword = readPropertyValue(propertyPrefix + "keystorePassword", String.class);
+            String keyPassword = readPropertyValue(propertyPrefix + "keyPassword", String.class);
+            String tlsProtocol = readPropertyValue(propertyPrefix + "tlsProtocol", String.class, "TLSv1.2");
+
+            if (keystore != null) {
+                try {
+                    SSLContext ctx = SSLContext.getInstance(tlsProtocol);
+                    KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+                    KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
+                    ks.load(new FileInputStream(keystore), keystorePassword.toCharArray());
+                    keyManagerFactory.init(ks, keyPassword.toCharArray());
+                    ctx.init(keyManagerFactory.getKeyManagers(), null, null);
+                    sslBuilder.context(ctx);
+                } catch (Exception e) {
+                    logger.error(e.getCause().toString());
+                    throw new IllegalStateException("Error creating the keystore for mongodb", e);
```

Comment:
Real diff is not that huge: 
![image](https://user-images.githubusercontent.com/4112568/162409310-2562f0e0-8f03-4be5-99ed-bba5b254e85f.png)